monitoring/gitlab.py

```diff
@@ -0,0 +1,137 @@
+"""Presentation Task that uploads Data and Plots to the EC-Earth 4 Gitlab."""
+
+import re
+import urllib.parse
+from pathlib import Path
+
+import jinja2
+from helpers.exceptions import PresentationException
+from helpers.presentation_objects import PresentationObject
+from scriptengine.exceptions import ScriptEngineTaskRunError
+from scriptengine.jinja import filters as j2filters
+from scriptengine.tasks.core import Task, timed_runner
+
+import gitlab
+
+
+class Gitlab(Task):
+    """Gitlab Presentation Task"""
+
+    _required_arguments = (
+        "src",
+        "local_dst",
+        "subject",
+        "template",
+        "api_key",
+    )
+
+    def __init__(self, arguments=None):
+        Gitlab.check_arguments(arguments)
+        super().__init__(arguments)
+
+    @timed_runner
+    def run(self, context):
+        sources = self.getarg("src", context)
+        dst_folder = self.getarg("local_dst", context)
+        issue_subject = self.getarg("subject", context, parse_yaml=False)
+        template_path = self.getarg("template", context)
+        key = self.getarg("api_key", context)
+        self.log_info(f"Create Gitlab issue '{issue_subject}'.")
+        self.log_debug(f"Template: {template_path}, Source File(s): {sources}")
+
+        presentation_list = self.get_presentation_list(sources, dst_folder)
+        gitlab_template = self.get_template(context, template_path)
+        gitlab_template.globals["urlencode"] = urllib.parse.quote
+
+        server_url = "https://gitlab.lrz.de"
+        gl = gitlab.Gitlab(server_url, private_token=key)
+
+        self.log_debug("Connecting to Gitlab.")
+
+        project, issue = self.get_project_and_issue(gl, issue_subject)
+
+        self.log_debug("Uploading attachments.")
+        issue.uploads = []
+        for item in presentation_list:
+            if item["presentation_type"] == "image":
+                file_name = Path(item["path"]).name
+                upload = project.upload(file_name, filepath=f"{dst_folder}/{file_name}")
```

Comment:
In Gitlab, we can only upload new files... As far as I can see, the API does not allow to update an existing file. I have tried to do this with the "update project files" functionality described [here](https://python-gitlab.readthedocs.io/en/stable/gl_objects/projects.html#id7), but this only works for actual repository files, i.e., you have to specify a branch/commit etc.

Reply:
This might become an issue, as it could fill the project space with quite many images. Without having looked into the API in detail, is there call to delete files? So we can mimic an update by delete+upload?

Reply:
update and delete only work for project files, so files actually lying in the repository. Actually, it's the same on Gitlab in the browser: It is not possible to manually delete an uploaded attachment to an issue/issue comment, either. I wonder if Gitlab does some cleanup automatically if an uploaded issue attachment is not referenced anywhere?

Reply:
Found [this](https://forum.gitlab.com/t/delete-files-uploaded-via-api/30305) but it doesn't help much. Searching the Gitlab [Project REST API](https://docs.gitlab.com/ee/api/projects.html) I didn't find anything to delete uploaded files. So it may not be a limitation just of python-gitlab.

Reply:
See the discussion on this in #89